db.py

```diff
@@ -1,87 +1,509 @@
-import copy
+from abc import ABC, abstractmethod
+import base64
+import binascii
+from collections.abc import Callable, Collection, Mapping, Sequence
+import json
+import logging
+import sqlite3
+from types import MappingProxyType
+
+import bcrypt
+import pypika  # type: ignore
+import pypika.terms
+
+
+logger = logging.getLogger(__name__)
+
 DB_PROPERTY = {
-    'users': {
-        'user_id': {'dtype':'int', 'auto_increment':True},
-        'username': {'dtype': 'str'},
-        'password': {'dtype': 'str'},
-        'permission': {'dtype': 'int'},
-        'auth_method': {'dtype': 'str'},
+    "users": {
+        "columns": {
+            "user_id": {"dtype": "int", "primary": True, "auto_increment": True},
+            "username": {"dtype": "str"},
+            "password": {"dtype": "str"},
+            "permission": {"dtype": "int"},
+            "auth_method": {"dtype": "str"},
+        },
     },
-    'symptoms': {
-        'symptoms_id': {'dtype': 'int', 'auto_increment': True},
-        'name': {'dtype': 'str'},
-        'academic': {'dtype': 'str'},
-        'visit': {'dtype': 'int'},
+    "symptoms": {
+        "columns": {
+            "symptoms_id": {"dtype": "int", "primary": True, "auto_increment": True},
+            "name": {"dtype": "str"},
+            "academic": {"dtype": "str"},
+            "visit": {"dtype": "int"},
+        },
     },
-    'bulletins': {
-        'bulletin_id': {'dtype': 'int', 'auto_increment': True},
-        'class': {'dtype': 'str'},
-        'user_id': {'dtype': 'int'},
-        'title': {'dtype': 'str'},
-        'content': {'dtype': 'str'},
-        'update_at': {'dtype': 'str'},
-        'create_at': {'dtype': 'str'},
+    "bulletins": {
+        "columns": {
+            "bulletin_id": {"dtype": "int", "primary": True, "auto_increment": True},
+            "class": {"dtype": "str"},
+            "user_id": {"dtype": "int"},
+            "title": {"dtype": "str"},
+            "content": {"dtype": "str"},
+            "update_at": {"dtype": "str"},
+            "create_at": {"dtype": "str"},
+        },
+    },
+    "clinics": {
+        "columns": {
+            "clinic_id": {"dtype": "int", "primary": True, "auto_increment": True},
+            "title": {"dtype": "str"},
+            "address": {"dtype": "str"},
+            "tel": {"dtype": "str"},
+            "tags": {"dtype": "array"},
+            "owner_id": {"dtype": "int"},
+        },
     },
-    'clinics': {
-        'clinic_id': {'dtype': 'int', 'auto_increment': True},
-        'title': {'dtype': 'str'},
-        'address': {'dtype': 'str'},
-        'tel': {'dtype': 'str'},
-        'tags': {'dtype': 'array'},
-        'owner_id': {'dtype': 'int'},
-    }
 }
```

Comment:
This is the updated schema specification format (the format we parse from). The schema itself is unchanged.

I gave up creating the table using the exported SQL format (the SQL commands exported from the database diagram), since it was not SQLite-compatible nor a fit for the features we needed.